### DISCUSSION TITLE
Error caused by calling CRF block

### DISCUSSION THREAD

**Discussion by sduqcj** [Project-MONAI/MONAI#4953] [Q&A]
（1）I introduced the CRF block at the end of the Unet framework:
def forward(self, x: torch.Tensor) -> torch.Tensor:
        
        x1 = self.model(x)       

        return CRF()(x,x1)

（2）I want to introduce CRF bloack at the end of the model for post-processing, but the following error is displayed:
monai.utils.module.OptionalImportError: import monai._C (No module named 'monai._C').

（3）After debugging, it is found that CRF module calls the filter function written in C language
import torch

from monai.utils.module import optional_import

_C, _ = optional_import("monai._C")

__all__ = ["BilateralFilter", "PHLFilter"]

There is a problem here. I don't know how to deal with it.

**Comment by wyli** [comment]
please try the docker image which has a precompiled `monai._C` in it and the CRF is probably ready-to-use if you can get the docker container running https://docs.monai.io/en/latest/installation.html#from-dockerhub

otherwise you can install it from the source: 
```
BUILD_MONAI=1 pip install --no-build-isolation git+https://github.com/Project-MONAI/MONAI#egg=monai
```

more details https://docs.monai.io/en/latest/installation.html#from-github

**Comment by sduqcj** [comment]
Thank you very much for your reply. I will try again according to your reply.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Project-MONAI/MONAI"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年8月22日(星期一) 晚上9:15
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Project-MONAI/MONAI] Error caused by calling CRF block (Discussion #4953)





 
please try the docker image which has a precompiled monai._C in it and the CRF is probably ready-to-use if you can get the docker container running https://docs.monai.io/en/latest/installation.html#from-dockerhub
 
otherwise you can install it from the source:
 BUILD_MONAI=1 pip install --no-build-isolation git+https://github.com/Project-MONAI/MONAI#egg=monai  
more details https://docs.monai.io/en/latest/installation.html#from-github
